designs/2024-support-ts-config-files/README.md

```diff
@@ -0,0 +1,130 @@
+- Repo: eslint/eslint
+- Start Date: 2024-03-09
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: [Arya Emami](https://github.com/aryaemami59)
+
+# Add Support for TypeScript Config Files
+
+## Summary
+
+Add support for TypeScript config files (`eslint.config.ts`, `eslint.config.mts`, `eslint.config.cts`)
+
+<!-- One-paragraph explanation of the feature. -->
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+The primary motivation for adding support for TypeScript configuration files to ESLint is to enhance the developer experience and accommodate the evolving JavaScript ecosystem. As TypeScript's popularity continues to grow, more projects are adopting TypeScript not only for their source code but also for their configuration files. This shift is driven by TypeScript's ability to provide compile-time type checks and IntelliSense. By supporting `eslint.config.ts`, `eslint.config.mts`, and `eslint.config.cts`, ESLint will offer first-class support to TypeScript users, allowing them to leverage these benefits directly within their ESLint configuration.
+
+## Detailed Design
```

Comment:
How does the feature interact with the [CLI option](https://eslint.org/docs/latest/use/command-line-interface#options) `--config`?

Reply:
I have tested it, so far it seems to work pretty well actually, especially with v9. I'm probably going to write a bunch of tests as well to see if there are any edge cases but so far so good.

Reply:
It would be good to explain what the behavior is when specifying TS or non-TS config files of varying file extensions through that option.

Reply:
It doesn't behave any differently, same as before. You can do `eslint . --config=eslint.config.ts` or `eslint . -c eslint.config.ts` and they just work. Same as with a `eslint.config.js` file.

Reply:
Can you add that into the RFC?

Reply:
I added it to the open questions, is that fine?